apps/meteor/server/models/raw/LivechatRooms.ts

```diff
@@ -2563,6 +2563,14 @@ export class LivechatRoomsRaw extends BaseRaw<IOmnichannelRoom> implements ILive
 			.toArray();
 	}
 
+	async countLivechatRoomsByPriority(): Promise<number> {
+		return this.col.countDocuments({ priorityWeight: { $exists: true, $ne: 99 } });
+	}
+
+	countLivechatRoomsWithDepartment(): Promise<number> {
+		return this.col.countDocuments({ departmentId: { $exists: true }, t: 'l' });
```

Comment:
Curious: is this using the partialFilterExp index or the sparse index on `departmentId`?

Reply:
I'll check

Reply:
The new approach (using priorityId) uses the sparse index...

```js
{ key: { priorityId: 1 }, sparse: true },
```

Reading  about Partitioned Filter Expression (PFE), seems that make more sense for this field, WDYT?

Reply:
You can give it a try, check times and see if it's worth it to have a new index there 👀